engine/c.cpp

```diff
@@ -214,21 +215,28 @@ KVDKStatus KVDKSet(KVDKEngine* engine, const char* key, size_t key_len,
 }
 
 KVDKStatus KVDKModify(KVDKEngine* engine, const char* key, size_t key_len,
-                      char** new_value, size_t* new_value_len,
-                      ModifyFunc modify, void* modify_args,
+                      KVDKModifyFunc modify_func, void* modify_args,
+                      DeallocateFunc modify_deallocate,
```

Comment:
Name as free_func()?
Since user have to pass this modify_deallocate in, maybe we should consider pass nv and nv_len out directly?